crates/rolldown_binding/src/options/plugin/js_plugin.rs

```diff
@@ -538,6 +538,10 @@ impl Plugin for JsPlugin {
   fn close_watcher_meta(&self) -> Option<rolldown_plugin::PluginHookMeta> {
     self.close_watcher_meta.as_ref().map(Into::into)
   }
+
+  fn register_hook_usage(&self) -> HookUsage {
+    HookUsage::all()
```

Comment:
That's so nice we could dynamically return if a hook is enabled depending on passed `JsPluginOptions`.

Reply:
yep